cub/benchmarks/bench/transform/babelstream.h

```diff
@@ -0,0 +1,75 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <cub/device/dispatch/dispatch_transform.cuh>
+#include <cub/util_namespace.cuh>
+
+#include <nvbench_helper.cuh>
+
+template <typename... RandomAccessIteratorsIn>
+#if TUNE_BASE
+using policy_hub_t = cub::detail::transform::policy_hub<false, ::cuda::std::tuple<RandomAccessIteratorsIn...>>;
+#else
+struct policy_hub_t
+{
+  struct max_policy : cub::ChainedPolicy<350, max_policy, max_policy>
+  {
+    static constexpr int min_bif    = cub::detail::transform::arch_to_min_bif(__CUDA_ARCH_LIST__);
+    static constexpr auto algorithm = static_cast<cub::detail::transform::Algorithm>(TUNE_ALGORITHM);
+    using algo_policy               = ::cuda::std::_If<
+                    algorithm == cub::detail::transform::Algorithm::fallback_for,
+                    void,
+                    ::cuda::std::_If<algorithm == cub::detail::transform::Algorithm::prefetch,
+                                     cub::detail::transform::prefetch_policy_t<TUNE_THREADS>,
+                                     ::cuda::std::_If<algorithm == cub::detail::transform::Algorithm::unrolled_staged,
+                                                      cub::detail::transform::unrolled_policy_t<TUNE_THREADS, TUNE_ITEMS_PER_THREAD>,
+                                                      cub::detail::transform::async_copy_policy_t<TUNE_THREADS>>>>;
+  };
+};
+#endif
+
+#ifdef TUNE_T
+using element_types = nvbench::type_list<TUNE_T>;
+#else
+using element_types =
+  nvbench::type_list<std::int8_t,
+                     std::int16_t,
+                     float,
+                     double
+#  ifdef NVBENCH_HELPER_HAS_I128
+                     ,
+                     __int128
+#  endif
+                     >;
+#endif
+
+// BabelStream uses 2^25, H200 can fit 2^31 int128s
+// 2^20 chars / 2^16 int128 saturate V100 (min_bif =12 * SM count =80)
+// 2^21 chars / 2^17 int128 saturate A100 (min_bif =16 * SM count =108)
+// 2^23 chars / 2^19 int128 saturate H100/H200 HBM3 (min_bif =32or48 * SM count =132)
+// inline auto array_size_powers = std::vector<nvbench::int64_t>{28};
+inline auto array_size_powers = nvbench::range(16, 28, 4);
+
+template <typename OffsetT, typename... RandomAccessIteratorsIn, typename RandomAccessIteratorOut, typename TransformOp>
```

Comment:
suggestion: you are tuning for `OffsetT=size_t` but use int at API entry point. This is not exactly sound, since different offset types might lead to quite different perf. I doubt that it's a source of huge difference for transform, but to be on the safe side I'd recommend introducing an `OffsetT{ct}` axis in transform benchmarks. That'd also make a bit more fair comparison with the Thrust baseline that'd switch to int for problem sizes under 2^31.

Reply:
Added an offset type axis. Thanks for the suggestion!